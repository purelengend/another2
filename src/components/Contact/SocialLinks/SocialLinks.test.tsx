import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import SocialLink from '.'

describe('SocialLink component', () => {
  const props = {
    href: '/',
    icon: 'icon-url',
    alt: 'Social Icon',
  }

  it('renders link with correct attributes', () => {
    const { container, getByRole } = render(
      <MemoryRouter>
        <SocialLink {...props} />
      </MemoryRouter>
    )

    const linkElement = getByRole('link', { name: props.alt })

    expect(linkElement).toBeInTheDocument()
    expect(linkElement).toHaveAttribute('href', props.href)
    expect(container).toMatchSnapshot()
  })

  it('renders image with correct attributes', () => {
    const { getByAltText } = render(
      <MemoryRouter>
        <SocialLink {...props} />
      </MemoryRouter>
    )

    const imageElement = getByAltText(props.alt)

    expect(imageElement).toBeInTheDocument()
    expect(imageElement).toHaveAttribute('src', props.icon)
  })
})
