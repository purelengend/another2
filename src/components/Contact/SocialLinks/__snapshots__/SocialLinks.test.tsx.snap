// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLink component renders link with correct attributes 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="Social Icon"
      class="chakra-image css-eqjwk7"
      src="icon-url"
    />
  </a>
</div>
`;
